lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/HostManager/HostManager.h"
+#include "glow/Backends/DeviceManager.h"
+#include "glow/Graph/Context.h"
+#include "glow/Runtime/RuntimeTypes.h"
+// include Partitioner
+
+#include <future>
+#include <queue>
+
+using namespace glow;
+using namespace runtime;
+
+HostManager::HostManager(std::vector<DeviceConfig> &configs) {
+  DeviceIDTy deviceCount = 0;
+  for (auto config : configs) {
+    auto newDevice = DeviceManager.getDevice(config);
+    devices_.emplace(deviceCount, newDevice);
+    deviceCount++;
+  }
+  provisioner_ = Provisioner();
+}
+
+HostManager::~HostManager() { clearHost(); }
+ResultCode HostManager::addNetwork(Module *M) {
```

Comment:
Currently HostManager does not take ownership of the Module, this can be changed if we desire. networks_ contains the DAGNodes which contain the runtime information but not the module, constants, or the compiled function.